src/librustdoc/json/conversions.rs

```diff
@@ -0,0 +1,606 @@
+            ImportItem(i) => ItemEnum::ImportItem(i.into()),
+            StructItem(s) => ItemEnum::StructItem(s.into()),
+            UnionItem(u) => ItemEnum::StructItem(u.into()),
+            StructFieldItem(f) => ItemEnum::StructFieldItem(f.into()),
+            EnumItem(e) => ItemEnum::EnumItem(e.into()),
+            VariantItem(v) => ItemEnum::VariantItem(v.into()),
+            FunctionItem(f) => ItemEnum::FunctionItem(f.into()),
+            ForeignFunctionItem(f) => ItemEnum::FunctionItem(f.into()),
+            TraitItem(t) => ItemEnum::TraitItem(t.into()),
+            TraitAliasItem(t) => ItemEnum::TraitAliasItem(t.into()),
+            MethodItem(m) => ItemEnum::MethodItem(m.into()),
+            TyMethodItem(m) => ItemEnum::MethodItem(m.into()),
+            ImplItem(i) => ItemEnum::ImplItem(i.into()),
+            StaticItem(s) => ItemEnum::StaticItem(s.into()),
+            ForeignStaticItem(s) => ItemEnum::StaticItem(s.into()),
+            ForeignTypeItem => ItemEnum::ForeignTypeItem,
+            TypedefItem(t, _) => ItemEnum::TypedefItem(t.into()),
+            OpaqueTyItem(t) => ItemEnum::OpaqueTyItem(t.into()),
+            ConstantItem(c) => ItemEnum::ConstantItem(c.into()),
+            MacroItem(m) => ItemEnum::MacroItem(m.source),
+            ProcMacroItem(m) => ItemEnum::ProcMacroItem(m.into()),
+            AssocConstItem(t, s) => ItemEnum::AssocConstItem { type_: t.into(), default: s },
+            AssocTypeItem(g, t) => ItemEnum::AssocTypeItem {
+                bounds: g.into_iter().map(Into::into).collect(),
+                default: t.map(Into::into),
+            },
+            StrippedItem(inner) => (*inner).into(),
+            _ => panic!("{:?} is not supported for JSON output", item),
+        }
+    }
+}
+
+impl From<clean::Module> for Module {
+    fn from(module: clean::Module) -> Self {
+        Module {
+            is_crate: module.is_crate,
+            items: module.items.into_iter().map(|i| i.def_id.into()).collect(),
+        }
+    }
+}
+
+impl From<clean::Struct> for Struct {
+    fn from(struct_: clean::Struct) -> Self {
+        let clean::Struct { struct_type, generics, fields, fields_stripped } = struct_;
+        Struct {
+            struct_type: struct_type.into(),
+            generics: generics.into(),
+            fields_stripped,
+            fields: fields.into_iter().map(|i| i.def_id.into()).collect(),
+            impls: Vec::new(), // Added in JsonRenderer::item
+        }
+    }
+}
+
+impl From<clean::Union> for Struct {
+    fn from(struct_: clean::Union) -> Self {
+        let clean::Union { struct_type, generics, fields, fields_stripped } = struct_;
+        Struct {
+            struct_type: struct_type.into(),
+            generics: generics.into(),
+            fields_stripped,
+            fields: fields.into_iter().map(|i| i.def_id.into()).collect(),
+            impls: Vec::new(), // Added in JsonRenderer::item
+        }
+    }
+}
```

Comment:
These impls are exactly the same, which means that the JSON output doesn't distinguish between structs and unions. Is that intentional?